### ISSUE TITLE
VST3 "MakePresets" not showing up in Cubase

### ISSUE THREAD

**Issue by rexmartin158** [iPlug2/iPlug2#751]
**Describe the bug**
VST3 "MakePresets" are not showing up in Cubase.  VST3 preset list does appear in other DAWs (Reaper, Cakewalk).  

VST3 plugins built in iPlug1 with MakePresets **_do_** show a preset list in Cubase - so appears support for Cubase "canned presets" is missing/broken in iPlug2.  Cannot determine reason/cause.

**To Reproduce**
Build VST3 plugin with "MakePresets()" and test in Cubase.  No presets show up in plugin window.  

**Expected behaviour**
List of presets in DAW plugin menu.

**IMPORTANT DETAILS**
* What plug-in format does it relate to - VST3

**Additional context**
I have "VST3_PRESET_LIST" listed in <ALL_EXTRA_DEFS> section of project-win.props file and VST3_DEFS line of common-mac.xcconfig.  Presets do show up in some DAWs but not Cubase/Steinberg DAWs.

I have the latest iPlug2 file set.

**Comment by rexmartin158** [comment]
I found this code block in IPlugConstants.h:

```
enum EVST3ParamIDs
{
#ifdef IPLUG1_COMPATIBILITY
  kBypassParam = 'bpas',
  kPresetParam = 'prst',
  kMIDICCParamStartIdx
#else
  kBypassParam = 65536,
  kPresetParam, // not used unless baked in presets declared
  kMIDICCParamStartIdx
#endif
};
```
1) What is "IPLUG1_COMPATIBILITY"?  
2) There is no other code anywhere that I can find that sets a value to kPresetParam when MakePreset or MakePresetFromNamedParameter is being used however it appears kPresetParam is required to inform the host of the preset list in IPlugVST3_ControllerBase.h 

`unitInfo.programListId = kPresetParam;`

Not sure how VST3 presets are being generated in any DAW given this finding - but it does work in some, not Cubase.

**Comment by olilarkin** [comment]
IPLUG1_COMPATIBILITY should be defined if you are trying to make an iPlug2 plugin compatible with state from a plug-in previously released with iPlug1 

https://github.com/iPlug2/iPlug2/wiki/Compile-time-macros

VST3 "baked in presets" have always been a bit horrible. Maybe they removed them from Cubase. I'll investigate

**Comment by rexmartin158** [comment]
> IPLUG1_COMPATIBILITY should be defined if you are trying to make an iPlug2 plugin compatible with state from a plug-in previously released with iPlug1
> 
> https://github.com/iPlug2/iPlug2/wiki/Compile-time-macros

Thank you for that link and info.  Are there any side effects of #defining IPLUG1_COMPATIBILTY ?

> Maybe they removed them from Cubase. I'll investigate

I checked some of my iPlug1 VST3 plugins and their presets do show up in the same version of Cubase - so it seems to be something in iPlug2.

**Comment by olilarkin** [comment]
The most significant one is with AU automation, which is quantized in iPlug1 because a flag was not set

https://github.com/iPlug2/iPlug2/search?q=IPLUG1_COMPATIBILITY

**Comment by olilarkin** [comment]
@rexmartin158  please could you give this branch a try? https://github.com/iPlug2/iPlug2/tree/VST3/preset-list

**Comment by rexmartin158** [comment]
I’m on travel until Monday with no access - but looking forward to trying
it ASAP.

Thank you.

**Comment by rexmartin158** [comment]
> @rexmartin158 please could you give this branch a try? https://github.com/iPlug2/iPlug2/tree/VST3/preset-list

Yes!  It works in Cubase now.  Thank you!

I suspected something in the section "#ifdef VST3_PRESET_LIST" in IPlugVST3ControllerBase.h but had no idea 3 other VST3 files also were affected. 

Well done!

**Comment by olilarkin** [comment]
Addressed in #89a233f
